addons/account/tests/test_portal_attachment.py

```diff
@@ -119,35 +129,36 @@ def test_01_portal_attachment(self):
             json={
                 'params': {
                     'attachment_id': create_res['id'],
-                    'access_token': create_res['access_token'],
+                    "access_token": create_res["as_author_access_token"],
                 },
             },
         )
         self.assertEqual(res.status_code, 200)
         self.assertFalse(self.env['ir.attachment'].sudo().search([('id', '=', create_res['id'])]))
 
-        # Test attachment can't be removed if not "pending" state
+        # Test attachment can be removed with token if not "pending" state
         attachment = self.env['ir.attachment'].create({
             'name': 'an attachment',
-            'access_token': self.env['ir.attachment']._generate_access_token(),
         })
         res = self.url_open(
             url=f'{self.invoice_base_url}/mail/attachment/delete',
             json={
                 'params': {
                     'attachment_id': attachment.id,
-                    'access_token': attachment.access_token,
+                    "access_token": limited_field_access_token(
+                        attachment, "as_author_access_token"
+                    ),
                 },
             },
         )
         self.assertEqual(res.status_code, 200)
-        self.assertTrue(self.env['ir.attachment'].sudo().search([('id', '=', attachment.id)]))
```

Comment:
Note to future readers: the change is acceptable due to the previous check verifying an implementation detail.
We were simply not able to confirm with the generic token of attachment if the user had to rights or not to delete it, so we assumed it was only fine when it was pending.
With the limited token only given to author, we know it is allowed, no matter the state.